dockers/docker-lldp-sv2/lldpmgrd

```diff
@@ -185,37 +192,52 @@ class LldpManager(object):
     def run(self):
         """
         Subscribes to notifications of changes in the PORT table
-        of the Redis State database. 
-        Subscribe to STATE_DB - get notified of the creation of an interface
+        of the Redis Config/Application database.
+        Subscribe to APP_DB - get port oper status
         Subscribe to CONFIG_DB - get notified of port config changes
         Update LLDP configuration accordingly.
         """
         # Set select timeout to 10 seconds
         SELECT_TIMEOUT_MS = 1000 * 10
 
-        # Subscribe to PORT table notifications in the State DB
         sel = swsscommon.Select()
-        sst = swsscommon.SubscriberStateTable(self.state_db, swsscommon.STATE_PORT_TABLE_NAME)
-        sel.addSelectable(sst)
-        sst = swsscommon.SubscriberStateTable(self.config_db, swsscommon.CFG_PORT_TABLE_NAME)
-        sel.addSelectable(sst)
 
-        # Listen for changes to the PORT table in the STATE_DB and CONFIG_DB
+        # Subscribe to PORT table notifications in the Config DB
+        sst_confdb = swsscommon.SubscriberStateTable(self.config_db, swsscommon.CFG_PORT_TABLE_NAME)
+        sel.addSelectable(sst_confdb)
+
+        # Subscribe to PORT table notifications in the App DB
+        sst_appdb = swsscommon.SubscriberStateTable(self.appl_db, swsscommon.APP_PORT_TABLE_NAME)
+        sel.addSelectable(sst_appdb)
+
+        # Listen for changes to the PORT table in the CONFIG_DB and APP_DB
         while True:
             (state, c) = sel.select(SELECT_TIMEOUT_MS)
 
             if state == swsscommon.Select.OBJECT:
-                (key, op, fvp) = sst.pop()
-
-                fvp_dict = dict(fvp)
-
-                # handle creation
-                if op == "SET" and fvp_dict.get("state") == "ok":
-                    self.generate_pending_lldp_config_cmd_for_port(key)
-
-                # handle config change
-                if op in ["SET", "DEL"] and (fvp_dict.get("alias") or fvp_dict.get("description")) :
-                    self.generate_pending_lldp_config_cmd_for_port(key)
+                (key, op, fvp) = sst_confdb.pop()
+                if fvp:
+                    fvp_dict = dict(fvp)
+
+                    # handle config change
+                    if (fvp_dict.has_key("alias") or fvp_dict.has_key("description")):
+                        if op in ["SET", "DEL"]:
+                            if self.is_port_up(key):
+                                self.generate_pending_lldp_config_cmd_for_port(key)
+                            else:
+                                self.pending_cmds.pop(key, None)
+
+                (key, op, fvp) = sst_appdb.pop()
+                if (key != "PortInitDone") and (key != "PortConfigDone"):
+                    if fvp:
+                        fvp_dict = dict(fvp)
+
+                        # handle port status change
+                        if fvp_dict.has_key("oper_status"):
+                            if "up" in fvp_dict.get("oper_status"):
```

Comment:
I think we should compare here `"up" == fvp_dict.get("oper_status")`
Also let's use one if here. Usually we write check if key is present and check the key as
`if "oper_status" in fvp_dict and fvp_dict["oper_status"] == "up"`

Reply:
In this case, though, we would need to do the `has_key` check again in the else statement. I think for a situation like this the nested if/else is better.

Reply:
This will result in a keyerror if oper_status is not present in fvp_dict.

---
In reply to: [304132193](https://github.com/Azure/sonic-buildimage/pull/3152#discussion_r304132193) [](ancestors = 304132193)

Reply:
@sumukhatv: It shouldn't, because the first condition will return false, so the second won't run.

However, you'd also need to do the same to the else case, so I think it's less efficient in this case.